packages/round-manager/src/features/round/__tests__/RoundApplicationForm.test.tsx

```diff
@@ -419,6 +419,60 @@ describe("Application Form Builder", () => {
       });
     });
   });
+
+  describe("Remove question", () => {
+    it("displays remove icon for each editable question", () => {
+      const editableQuestions = initialQuestions;
+      renderWithContext(
+        <RoundApplicationForm
+          initialData={{
+            program: {
+              operatorWallets: [],
+            },
+          }}
+          stepper={FormStepper}
+        />
+      );
+
+      expect(screen.getAllByTestId("remove-question")).toHaveLength(
+        editableQuestions.length
+      );
+    });
+
+    it("removes question when remove icon is clicked", () => {
+      const editableQuestions = initialQuestions;
+
+      const indexToBeRemoved = randomInt(0, 3);
+
+      renderWithContext(
+        <RoundApplicationForm
+          initialData={{
+            program: {
+              operatorWallets: [],
+            },
+          }}
+          stepper={FormStepper}
+        />
+      );
+
+      expect(
+        screen.getAllByTestId(`field-${indexToBeRemoved}`)
+      ).toHaveLength(1);
+
+      const removeIcons = screen.getAllByTestId("remove-question");
+      fireEvent.click(removeIcons[indexToBeRemoved]);
+
+      expect(
+        screen.getAllByTestId("remove-question")
+      ).toHaveLength(editableQuestions.length - 1);
+
+      // THIS IS FAILING
+      expect(
```

Comment:
@shavinac / @dschinkel this is failing which is confusing.
The UI flow works as expected but I'm assuming this isn't the right way to check if the right element is removed